game/src/game/player/mod.rs

```diff
@@ -1552,3 +1576,67 @@ fn track_hits(
         }
     }
 }
+
+fn update_serpentring_health(
+    query: Query<(&Passives, Option<&Grounded>), With<Player>>,
+    mut health_q: Query<&mut Health, With<Player>>,
+    config: Res<PlayerConfig>,
+) {
+    if let Ok((passives, grounded)) = query.get_single() {
```

Comment:
Why are we checking for grounded here? And why two queries above? You should query for (&mut Health, &Passives), With<Player> in one query.
Also instead of running this all the time it should probably be run only when the passive is picked up right?